### DISCUSSION TITLE
BRT-100-TRV temperature accuracy

### DISCUSSION THREAD

**Discussion by michal-bartak** [Koenkk/zigbee2mqtt#9155] [Devices]
I just bought a few Moes BRT-100-TRVs. Paired with z2m without problems (HA 1.9.7 with Z2M 1.21.2-1)
I'm just learning them. What I havever early noticed is, that Z2M exposes temperature without fraction of degrees. 

Manual of TRV says it provides 0.5 degree of Celsius accuracy. Also the TRV itself is able present half of degree of Celsius on its display for both: measuring and setting target temperature

Is there any chance to get such accuracy in Z2M?

**Comment by Rigolman69** [comment]
Hello,

Do you still carry on experience this product ?
I just get one of them. I am not fully aware about it but it seems to look fine.
Regarding you request accuracy is +/- 0.5°C. But it is not actually related to what you see on app. Accuracy is not what you see. What you see is temperature your app as acquired. In my app I don't see neither decimals expect in historic :

![Capture d'écrans_20220102-133315](https://user-images.githubusercontent.com/96980371/147875843-c1527f23-3b0d-4ba9-a9de-b82f379049f1.png)

**Reply by michal-bartak** [reply]
What app you are using?
I'm using Home Assistant. And looking at history I still can see 1 degree resolution. For all temperatures: reported, calibration offset, min and max settings etc. 
Looks like it is how zigbee2mqtt reports it. 

What I said is, that at least TRV itself shows temperatures (and allows to set temperature setpoints) with 0.5 degree resolution.

Here is github issue I reported, but a its developer says it's limitation of those TRVs. I'm also surprised you are getting .75 fraction of degree. The developer says it can be 0.5 only.
https://github.com/Koenkk/zigbee2mqtt/issues/9158

**Reply by Rigolman69** [reply]
I use JEEDOM, French app : https://github.com/Koenkk/zigbee2mqtt/discussions
Looking at the logs, I can see 0.5°C resolution :
![Capture d'écrans_20220102-212954](https://user-images.githubusercontent.com/96980371/147888898-0b9b6114-1e5b-4e35-a2af-c41045818641.png)

When I see XX.75°C or else value, I guess it's because of average process in historic.

**Reply by Vodny86** [reply]
Dood day.
Did you solve fractions of degrees? I have a same problem. I can set/read only whole degree.

Thank you.

**Comment by Ivano62** [comment]
I confirm, with HA and z2m I can set only integer values
